server/datastore/mysql/queries.go

```diff
@@ -298,16 +282,23 @@ func (ds *Datastore) NewQuery(
 }
 
 func (ds *Datastore) updateQueryLabels(ctx context.Context, query *fleet.Query) error {
-	return ds.updateQueryLabelsInTx(ctx, query, nil)
+	err := ds.withRetryTxx(ctx, func(tx sqlx.ExtContext) error {
+		return ds.updateQueryLabelsInTx(ctx, query, tx)
+	})
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "updating query labels")
+	}
+	return nil
 }
 
 // updates the LabelsIncludeAny for a query, using the string value of
 // the label. Labels IDs are populated
-func (ds *Datastore) updateQueryLabelsInTx(ctx context.Context, query *fleet.Query, txToUse *sqlx.Tx) error {
-	var (
-		tx  *sqlx.Tx
-		err error
-	)
+func (ds *Datastore) updateQueryLabelsInTx(ctx context.Context, query *fleet.Query, tx sqlx.ExtContext) error {
+	if tx == nil {
+		return ctxerr.New(ctx, "updateQueryLabelsInTx called with nil tx")
+	}
+
+	var err error
```

Comment:
Not starting our own transaction in this method greatly simplifies it, as an added bonus.